### PR TITLE
kubeadm cert documentation

### PR DESCRIPTION
**Pull request by liztio**
Fixes kubernetes/kubeadm#1242

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+### certs renew
+
+Certificates can also be renewed using `kubeadm alpha certs renew --use-csr`. 
+Like with `kubeadm init`, an output directory can be specified with `--csr-dir`. 
+To use the new certificates, copy the signed certificate and private key into the PKI directory (by default `/etc/kubernetes/pki`)
+
+## Cert usage
+
+CSRs contain a certificate's name, domains, and IPs, but it does _not_ specify usages. 
+It is the responsibily of the CA to specify [the correct cert usages][cert-table] when issuing a certificate.
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: responsibility

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+To better integrate with external CAs, kubeadm can also produce certificate signing requests (CSRs). 
+A CSR represents a request for a CA to produce a signed certificate for a client. 
+In kubeadm terms, any certifate that would normally be signed by an on-disk CA can be produced as a CSR instead, but CAs themselves cannot.
+
+## Requesting CSRs
+
+### kubeadm init 
+You can request an individual CSR with `kubeadm init phase certs apiserver --use-csr`. 
+The `--use-csr` flag can only be applied to invidiual phases; once [all certificates are in place][certs] you can run `kubeadm init --external-ca`.
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: individual

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+[manage-tls]: https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/
+[cert-manager]: https://github.com/jetstack/cert-manager
+[certs]: https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#certificate
+
+# Using Certificate Requests with Kubeadm
+
+To better integrate with external CAs, kubeadm can also produce certificate signing requests (CSRs). 
+A CSR represents a request for a CA to produce a signed certificate for a client. 
+In kubeadm terms, any certifate that would normally be signed by an on-disk CA can be produced as a CSR instead, but CAs themselves cannot.
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: certificate

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
+---
+reviewers:
+- sig-cluster-lifecycle
+title: Certificate Management with kubeadm
+content_template: templates/task
+---
+
+{{% capture overview %}}
+
+This page explains various kubeadm-specific topics related to certificate management
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: i think this should be under our folder in setup/independent. WDYT?
the idea was to move all kubeadm pages to setup/ eventually, except the reference docs.

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+Kubeadm can renew certificates using the `kubeadm alpha certs renew` commands. 
+Typically this is done by loading on-disk CA certificates and keys and using them to issue new certificates.
+This works fine when using an entirely self-contained certificate tree, but may not make sense when certificates are managed externally.
+
+As an alternative, kubernetes provides its own [API for managing certificates][manage-tls].
+You can direct `kubeadm` to use this API with `kubeadm alpha certs renew --use-api`.
+
+## Setting up a signer
+
+The kubernetes certificate authority does not work out of the box.
```
Comment (neolit123) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: Kubernetes uppercase

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md (hunk 4 above)
Comment (liztio) [reply content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: This was the location @Bradamant3 and I agreed on. `--use-api`, for example, isn't setup-related, it's only ever used after a cluster has been set up and running for a while.

**Comment by liztio** [comment]
@Bradamant3 I noticed that the styling of `<h1>` is off, there's no margin at the top. Should I only be using h2 and below in my page?

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+A CSR represents a request for a CA to produce a signed certificate for a client. 
+In kubeadm terms, any certificate that would normally be signed by an on-disk CA can be produced as a CSR instead, but CAs themselves cannot.
+
+## Requesting CSRs
+
+### kubeadm init 
+You can request an individual CSR with `kubeadm init phase certs apiserver --use-csr`. 
+The `--use-csr` flag can only be applied to individual phases; once [all certificates are in place][certs] you can run `kubeadm init --external-ca`.
+
+You can pass in a directory to output the CSRs to with `--csr-dir`.
```
Comment (shavidissa) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: You can pass in a directory to output the CSRs to with  - > Remove the extra 'to' before with. :)

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+In kubeadm terms, any certificate that would normally be signed by an on-disk CA can be produced as a CSR instead, but CAs themselves cannot.
+
+## Requesting CSRs
+
+### kubeadm init 
+You can request an individual CSR with `kubeadm init phase certs apiserver --use-csr`. 
+The `--use-csr` flag can only be applied to individual phases; once [all certificates are in place][certs] you can run `kubeadm init --external-ca`.
+
+You can pass in a directory to output the CSRs to with `--csr-dir`. 
+If it is not specified, the default certificate directory (`/etc/kubernetes/pki`) will be used.
```
Comment (shavidissa) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: If it is not specified, the default certificate directory (`/etc/kubernetes/pki`) will be used. - > If it is not specified, the default certificate directory (`/etc/kubernetes/pki`) is used. 

(Keeping the sentece in the present tense.)

File: content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+Both the CSR and the accompanying private key will be output there. Once a certificate is signed, both it and the private key must be copied to the PKI directory (by default `/etc/kubernetes/pki`).
```
Comment (shavidissa) [inline content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-certs.md]: Minor suggestion: :)
Both the CSR and the accompanying private key are given in the output. Once a certificate is signed, the certificate and the private key must be copied to the PKI directory (by default `/etc/kubernetes/pki`).

**Comment by liztio** [comment]
@shavidissa updated!

**Comment by timothysc** [comment]
@kubernetes/sig-docs-pr-reviews  

This is ready to go!

**Pull request merged**